Sources/Amplitude/Amplitude.swift

```diff
@@ -39,6 +41,9 @@ public class Amplitude {
         let contextPlugin = ContextPlugin()
         self.contextPlugin = contextPlugin
 
+        (self.storage as? PersistentStorage)?.amplitude = self
```

Comment:
@izaaz Yeah, found there is a reference to Amplitude instance while I didn't find a place that passed it in. So I can't connect to it. Similar for the current logger in storage, which seems not connected as well. I could change to initiate a Diagonostics object in storage then let the amplitude diagnostic refer that. @justin-fiedler Do you know if we are setting amplitude instance any where else or if I missed anything for that?

Reply:
I would also prefer to keep the dependencies in Storage to a minimum (i.e. only pass in diagnostics/logger).

What about making diagnostics part of the configuration so that it can be passed into the default instances of storage?

Reply:
That's a good idea, I just modified the persistence storage constructor to take a logger and diagnostic and making diagnostics part of the configuration.